magpie/base/document.py

```diff
@@ -25,7 +26,13 @@ def __init__(self, doc_id, filepath, text=None):
             self.filename = os.path.basename(filepath)
 
             with open(filepath, 'r') as f:
-                self.text = f.read().decode('utf-8')
+                try:
+                    self.text = f.read().decode('utf-8')
+                except AttributeError:
+                    if sys.version_info.major == 3:
+                        self.text = f.read()
+                    else:
+                        raise
```

Comment:
yep it would i did the quick n dirty :D as i was trying to get this working for a project I'm working on.